src/dist/dist.rs

```diff
@@ -543,6 +568,35 @@ impl ToolchainDesc {
         (channels.iter().any(|x| *x == self.channel) || TRACKING_VERSION.is_match(&self.channel))
             && self.date.is_none()
     }
+
+    pub(crate) fn resolve_delta(channel: &str, delta: u64) -> Result<String> {
+        if channel != "stable" {
```

Comment:
If this is the only use of `channel`, I'd probably handle this condition outside of the function.

Reply:
The original intent was to leave room for other deltas (I don't know what `beta-2` means, but it could possibly mean something later on)... But indeed it could be a premature optimization.

Reply:
Yes, I'd simplify this for now -- can always be changed later.